packages/kilt-did/README.md

```diff
@@ -3,6 +3,320 @@
+
+// Ask the keystore to generate a new keypair to use for encryption.
+const encryptionKeyPublicDetails = await keystore.generateKeypair({
+  alg: EncryptionAlgorithms.NaclBox,
+  seed: encryptionSeed,
+})
+
+// Specify the endpoints the DID wishes to expose to other parties.
+const serviceEndpoints: IServiceDetails[] = [
+  {
+    id: 'email',
+    serviceEndpoint: 'my@email.org',
+    type: 'EmailService',
+  },
+]
+
+// Generate the KILT light DID with the information generated.
+const lightDID = new LightDidDetails({
+  authenticationKey: {
+    publicKey: authenticationKeyPublicDetails.publicKey,
+    type: authenticationKeyPublicDetails.alg,
+  },
+  encryptionKey: {
+    publicKey: encryptionKeyPublicDetails.publicKey,
+    type: encryptionKeyPublicDetails.alg,
+  },
+  services: serviceEndpoints,
+})
+
+// Will print `did:kilt:light:014sxSYXakw1ZXBymzT9t3Yw91mUaqKST5bFUEjGEpvkTuckar:omFlomlwdWJsaWNLZXlYID9hc7PRyRlUp+syykH3KrsVZlObWlfqtegO1KRzuo8zZHR5cGV4GHgyNTUxOS14c2Fsc2EyMC1wb2x5MTMwNWFzgaNiaWRlZW1haWxvc2VydmljZUVuZHBvaW50bG15QGVtYWlsLm9yZ2R0eXBlbEVtYWlsU2VydmljZQ==`.
+console.log(lightDID.did)
+```
+
+## Full DIDs
+
+As mentioned above, the creation of a full DID requires interaction with the KILT blockchain. Therefore, it is necessary for the DID creation operation to be submitted by a KILT address with enough funds to pay the transaction fees.
+
+By design, DID signatures and Substrate signatures are decoupled, meaning that the encoded and signed DID creation operation can then be signed and submitted by a different KILT account than the DID subject. We think this opens the path for a wider range of use cases in which, for instance, a service provider might be willing to offer a DID-as-a-Service option for its customers.
+
+An example of a full DID is the following:
+
+```
+did:kilt:4rp4rcDHP71YrBNvDhcH5iRoM3YzVoQVnCZvQPwPom9bjo2e
+```
+
+Here, there is no `light:` component, which indicates that the DID is a full DID and that the keys associated with it must not be derived from the DID identifier but must be retrieved from the KILT blockchain.
+
+Beyond an authentication key, and encryption key, and external services, a full DID also supports an **attestation key**, which must be used to write CTypes and attestations on the blockchain, and a **delegation key**, which must be used to write delegations on the blockchain.
+
+### Creating and anchoring a full DID
+
+The following is an example of how to create and write a full DID on chain that specifies only an authentication key.
+
+```typescript
+import Keyring from '@polkadot/keyring'
+
+import Kilt from '@kiltprotocol/sdk-js'
+import { BlockchainUtils } from '@kiltprotocol/chain-helpers'
+import {
+  DefaultResolver,
+  DemoKeystore,
+  DidUtils,
+  SigningAlgorithms
+} from '@kiltprotocol/did'
+import { KeyRelationship } from '@kiltprotocol/types'
+
+// Initialise connection to a KILT blockchain node.
+await Kilt.init({ address: 'ws://127.0.0.1:9944' })
```

Comment:
I guess people trying to write some code example would like a template that works for them. Do we expect they went beyond getting the package from NPM and actually started a local Kilt blockchain? If not, should the template rather reference one of the public endpoints?